### DISCUSSION TITLE
GitHub Action 同步 Notion 无法拉取 HTML 格式

### DISCUSSION THREAD

**Discussion by zixiangcode** [LetTTGACO/elog#62] [Q&A]
我尝试采用 GitHub Action 来自动化备份我的 Notion 文章。
下面是我的 `elog.config.js` 中的配置**片段**：
```js
module.exports = {
  deploy: {
    platform: 'local',
    local: {
      outputDir: './article',
      filename: 'title',
      format: 'html',
      catalog: true,
    },
  }
```
在这里，对于 `format` 字段的取值是 `html`，但是 `elog sync` 拉取到的文章仍然是 Markdown 格式的文章。请问 Elog 能拉取 Notion 中的静态 HTML 吗？如果能的话，是我的配置哪里有问题吗？还是说 Elog 只能导出 Notion 的 md 格式啊？不是很清楚 `format` 字段的含义，还望解答！

**Comment by LetTTGACO** [comment]
从代码运行流程上来看，这是一个 bug，因为生成的本地文件后缀被我写死为`.md`了。当导出为本地时，无论选择什么样的 format 格式，本地文件都会以`.md`结尾，但是文件内容却是相应格式的。

从业务角度出发，导出为 html 目前还不完善，因为我所做的就是将 notion 文章下载为 markdown，然后再基于 markdown 转换为 html，是没有任何样式加持的，所以导出为 html 在导出到本地的场景中几乎不可用。当然这只是我认为的场景，如果你有相关的应用场景，可以和我讨论，目前 elog 也支持自定义文档处理器帮助html 文档添加样式

在未来的 1.x 版本中，导出为 html 会将完善起来，我会考虑添加一些样式主题和预览样式的功能，如果你不着急可以再等等。

**Reply by LetTTGACO** [reply]
如果你的需求是将 notion 的样式原封不动导出到本地，目前是不支持的

**Reply by zixiangcode** [reply]
Notion 的 Export 方式中有静态的 HTML 选项，所以 ”将 Notion 文章下载为 Markdown“ 是否可以改为下载为 HTML 呢（不是很清楚您项目背后的下载原理）？
有这个需求的原因主要在于，我发现将 Notion 的文章导出为 Markdown 的时候，格式会发生错误，尤其是行内的 Latex 公式的问题（当然可能是我的 Latex 代码没有输入正确），也有一些其他的 Notion 格式很不适合 Markdown 的展示。所以我才想能否备份静态的 HTML 文件出来。
Anyway，感谢您的答复和您的项目！

**Reply by LetTTGACO** [reply]
https://elog.1874.cool/notion/notion-example
「行内公式，在Vitepress会报错，不做演示」

你是否用的是 vitepress，如果是的话那就是 vitepress 默认主题不兼容显示，我本地测试也会报错。所以我就避免写行内公式改用块级公式。
当然你也可以考虑寻找支持行内公式的插件，我觉得应该是有的，或者用别的能兼容行内公式的 vitepress 主题

**Reply by LetTTGACO** [reply]
如果只是备份的要求，确实 Notion 的富文本格式丰富度是远大于 markdown 格式的。
Elog 在下载各个写作平台的文档时，都是以 markdown 格式为基准，如果有需要再转换成其他格式，所以本身就会存在样式丢失。
解决办法有两个：
1. 将 markdown 不支持的 notion 格式转成图片贴在 notion 文章中
2. 使用隔壁的 NotionNext，它是完完全全以Notion接口直接渲染为 Html 富文本，基本不会有样式损失，在大部分场景都做到了 Notion 样式的还原。

未来 Elog 转成完全插件形式的话，也可能会出现Notion直接导出富文本的插件，可以关注一下

**Reply by zixiangcode** [reply]
感谢您的耐心答复。
我的需求场景是通过 Elog + GitHub Action 来完成 Notion 指定数据库中的文章自动备份到 GitHub 中。所以我选择的 Elog 的部署平台就是本地部署。我想我出现的行内公式报错与您的 Elog 项目并无关系，因为我发现我直接从 Notion 中 Export Markdown 后，在我本机的 Typora 中阅读也会有公式显示的错误问题。所以我猜测行内公式显示的问题应该在于我书写了不标准的 Latex 公式？（虽然这个公式可以被 Notion 正确渲染）。
因此考虑到静态 HTML 就不会有这种问题，而且可以最大程度上保留 Notion 的原生样式。所以我期望 Elog 能够下载 Notion 导出的静态 HTML 格式作为备份。（Notion 本身是可以选择导出静态的 HTML 的）

**Reply by zixiangcode** [reply]
隔壁的 NotionNext 我也有在用，但是似乎没摸索出怎么配合 GitHub Action 完成自动备份静态的 HTML 文件（鄙人对前端了解不多，还请见谅）

**Reply by LetTTGACO** [reply]
那确实一部分的本地编辑器/markdown 渲染器 是不太兼容非标准的md 格式，可以考虑安装插件或者换一种写法来处理。
目前 Elog 用的 notion 转 md 也是第三方库[notion-to-md](https://github.com/souvikinator/notion-to-md)，所有的样式转换都是它处理的。如果有问题暂时只能等待库作者来解决

**Reply by LetTTGACO** [reply]
emmm 那确实目前还没有非常完美的解决方案，只能再等等了

**Reply by zixiangcode** [reply]
明白了，感谢您的答复。我最初以为 Elog 是利用 Notion 的 API 直接从 Notion 的 Export 里面直接下载 md 文件
